models/optional_modules/consent/snowplow_web_consent_users.sql

```diff
@@ -12,48 +12,42 @@ with base as (
 
   select
     domain_userid,
-    max(derived_tstamp) as last_processed_event,
+    user_id,
+    geo_country,
+    max(load_tstamp) as last_processed_event,
     count(case when event_name = 'cmp_visible' then 1 end) as cmp_events,
     count(case when event_name = 'consent_preferences' then 1 end) as consent_events,
-    max(case when event_name = 'cmp_visible' then derived_tstamp end) as last_cmp_event_tstamp
+    max(case when event_name = 'cmp_visible' then derived_tstamp end) as last_cmp_event_tstamp,
+    row_number() over(partition by domain_userid order by max(load_tstamp) desc) as latest_event_by_user_rank
 
   from {{ ref('snowplow_web_consent_log') }}
 
-  where
-  -- enough time has passed since the event happened (derived_tstamp) to be able to process it as a whole (please bear in mind the late arriving data)
-  cast({{ dbt_utils.dateadd('hour', var("snowplow__consent_late_arriving_buffer", 10), 'derived_tstamp') }} as {{ dbt_utils.type_timestamp() }}) < {{ dbt_utils.current_timestamp_in_utc() }}
-
   {% if is_incremental() %} -- and it has not been processed yet
-  and derived_tstamp > ( select max(last_processed_event) from {{ this }} )
+  where load_tstamp > ( select max(last_processed_event) from {{ this }} )
```

Comment:
🔧 It's a small table so it shouldn't matter, but is it worth sticking this in a set block above to avoid the subquery filter, just in case?